static/js/message_edit.js

```diff
@@ -125,45 +125,54 @@ export function is_message_editable_ignoring_permissions(message) {
     return true;
 }
 
-export function get_editability(message, edit_limit_seconds_buffer = 0) {
-    if (!is_message_editable_ignoring_permissions(message)) {
-        return editability_types.NO;
+function is_content_editable(message, edit_limit_seconds_buffer = 0) {
+    if (!page_params.realm_allow_message_editing) {
+        return false;
     }
 
-    if (!is_topic_editable(message, edit_limit_seconds_buffer)) {
-        return editability_types.NO;
+    if (!message.sent_by_me) {
+        return false;
     }
 
-    if (!page_params.realm_allow_message_editing) {
-        if (message.type === "stream") {
-            return editability_types.TOPIC_ONLY;
-        }
-        return editability_types.NO;
+    if (is_widget_message(message)) {
+        return false;
     }
 
-    if (
-        page_params.realm_message_content_edit_limit_seconds === null &&
-        message.sent_by_me &&
-        !is_widget_message(message)
-    ) {
-        return editability_types.FULL;
+    if (page_params.realm_message_content_edit_limit_seconds === null) {
+        return true;
     }
 
     if (
         page_params.realm_message_content_edit_limit_seconds +
             edit_limit_seconds_buffer +
             (message.timestamp - Date.now() / 1000) >
-            0 &&
-        message.sent_by_me &&
-        !is_widget_message(message)
+        0
     ) {
+        return true;
+    }
+    return false;
+}
+
+export function get_editability(message, edit_limit_seconds_buffer = 0) {
+    if (!is_message_editable_ignoring_permissions(message)) {
+        return editability_types.NO;
+    }
```

Comment:
I worry about `is_topic_editable` / `is_content_editable` not calling `is_message_editable_ignoring_permissions`; I feel like folks are going to call those functions directly and skip that check by accident. 

If it's really cheap, probably it'd be reasonable move the check into both functions? I wouldn't be surprised if that could be followed by a refactor that deletes `get_editability` in favor of calling those functions directly in most places -- the reason `get_editability` made sense as an interface was that there was a sort order to the possibilities.